src/daft-core/src/array/ops/list_agg.rs

```diff
@@ -98,6 +96,20 @@ impl DaftListAggable for crate::datatypes::PythonArray {
     }
 }
 
+impl DaftListAggable for ListArray {
+    type Output = DaftResult<ListArray>;
+
+    fn list(&self) -> Self::Output {
```

Comment:
Yes, all the nested types don't have an implementation yet because they were waiting on this ListType to be implemented. I'm writing a followup to do it once this stuff is in.